integr8ly/tutorial-web-app
# INTLY-2346 openshift 4 support

@philbrookes Would you mind having a quick review of this? I will also add OpenShift 4 verification steps in the morning.

I still need to confirm OpenShift 3 backwards compatibility is working as expected.

## OpenShift 4 Verification
- Have an existing OpenShift 4 cluster with Integreatly setup
- Run `oc status`, note the `api.<my-cluster>` URL
- In the solution explorer Namespace, update the `tutorial-web-app` DeploymentConfig:
  - Update the image to `docker.io/aidenkeatingrht/tutorial-web-app:INTLY-2346`
  - Add an env var called `OPENSHIFT_API` with the value of the `api.<my-cluster>` from earlier
  - Add an env var called `OPENSHIFT_VERSION` with the value `'4'`
- Wait for the DeploymentConfig to redeploy
- Wait for the webapp to become available, you can check the logs of the Pod
- Go into the webapp URL and start Walkthrough 1A (AMQ Online, Fuse Online)
- Ensure
  - Fuse Online starts provisioning
  -  AMQ Online starts provisioning
- Once the two services have provisioned, ensure
  - The walkthrough tasks show as expected
  - Fuse Online is accessible through the various links in the Walkthrough
  -  AMQ online is accessible through the various links in the Walkthrough
- Note that there will be missing crud app URL's as that relies on a separate PR being merged
- Go to the homepage of the webapp
- Ensure AMQ Online on the right-hand side shows itself as available, click it and ensure it brings you to the AMQ Online console

## OpenShift 3 Verification
- Have an existing OpenShift 3 cluster with Integreatly installed on it
- If using 1.4.1 release of Integreatly
  - Ensure managed-service-broker is updated to use tag greater or equal to `v0.0.9`
  - Ensure managed-service-broker has `USER_SSO_URL` env set to the user sso route
- Update the webapp to use image `docker.io/aidenkeatingrht/tutorial-web-app:INTLY-2346`
- Open the webapp, ensure services load as expected
- Do walkthrough 1A, ensure it works as expected
